ACMS-1312: Revamped the redirection code for site preview.

panshulK opened the pull request:
**Motivation**
* To remove the re-direction code from acquia_cms_headless module and move to acquia_cms_headless_ui module.
* To re-write the redirection logic to work only when destination params are not set.
* Loom recording - https://www.loom.com/share/8fc6be4203a6428c97c4f98d1b97bb39
Fixes #NNN

**Proposed changes**
* To remove the re-direction code from acquia_cms_headless module and move to acquia_cms_headless_ui module.
* To re-write the redirection logic to work only when destination params are not set.

**Alternatives considered**
* None

**Testing steps**
* Enable headless mode and head towards - /admin/content page.
* Edit any node and notice that node edit URL has the destination query params ex. `?destination=/acquia_cms/admin/content`
* Make changes and notice that the user is redirected to destination url from query params.
* Edit the content again, remove the query params completely and then save.
* Notice that user is redirected to `/node/<id>/site-preview` page.

Loom Recording - https://www.loom.com/share/8fc6be4203a6428c97c4f98d1b97bb39

**Merge requirements**
- [ ] _Major change_, _Minor change_, _Bug_, _Enhancement_, and/or _Chore_ label applied
- [ ] Manual testing by a reviewer

fiasco commented on modules/acquia_cms_headless/modules/acquia_cms_headless_ui/acquia_cms_headless_ui.module:
```diff
@@ -29,6 +31,20 @@ function acquia_cms_headless_ui_help($route_name) {
+  // If not set, then redirect to site-preview.
+  if (!$current_request->query->get('destination')) {
+    $response = new RedirectResponse($url->toString());
+    $response->send();
```
How is this going to respond in (a) bulk update scenarios and (b) over CLI?

fiasco replied:
This feels like an action we should take on the form rather than on hook_node_update.
This module also has a number or route overrides that maybe worth reviewing. Perhaps the better approach is to set the /node/<nid>/latest route to use entity.node.headless_preview controller?

vishalkhode1 commented on modules/acquia_cms_headless/modules/acquia_cms_headless_ui/acquia_cms_headless_ui.module:
```diff
@@ -96,6 +97,21 @@ function acquia_cms_headless_ui_form_node_form_alter(array &$form, FormStateInte
     ];
   }
   $form['#process'][] = 'acquia_cms_headless_ui_process_node_type_workflow_options';
+  $form['actions']['submit']['#submit'][] = 'acquia_cms_headless_ui_redirect_user_to_site_preview';
```
@panshulK I think the redirection of form is handling from function added above i.e 
`Redirect::entityForm($form, $form_state);`
and looks like handled from [here](https://github.com/acquia/acquia_cms/blob/develop/modules/acquia_cms_headless/modules/acquia_cms_headless_ui/src/Redirect.php#L68).
Please see.

panshulK replied:
Hey @vishalkhode1  I've updated the PR with changes discussed please review.

vishalkhode1 reviewed (changes requested):
@panshulK How does the Route Subscriber is getting called ? Are we missing `.serices.yml` file or we've added in some other PR ?

panshulK commented:
> @panshulK How does the Route Subscriber is getting called ? Are we missing `.serices.yml` file or we've added in some other PR ?
@vishalkhode1  surprisingly i could see the corresponding service.yml file changes when i switched to the file and they were not even showing as "Changes not staged for commit."
I removed the existing service.yml file on local, created new with same contents and then I could see my changes being reflected as "Ready to be staged for commit".

I have updated the PR with the corresponding service file. Please review and let me know if any other changes are required.

Thanks
